src/compiler/transpile/run-program.ts

```diff
@@ -43,16 +43,20 @@ export const runTsProgram = async (
   const typesOutputTarget = config.outputTargets.filter(isOutputTargetDistTypes);
   const emittedDts: string[] = [];
   const emitCallback: ts.WriteFileCallback = (emitFilePath, data, _w, _e, tsSourceFiles) => {
-    if (emitFilePath.endsWith('.js') || emitFilePath.endsWith('js.map')) {
-      updateModule(config, compilerCtx, buildCtx, tsSourceFiles[0], data, emitFilePath, tsTypeChecker, null);
-    } else if (
-      emitFilePath.endsWith('.e2e.d.ts') ||
-      emitFilePath.endsWith('.spec.d.ts') ||
+    if (
+      emitFilePath.includes('.e2e.') ||
+      emitFilePath.includes('/e2e.') ||
```

Comment:
I feel like this is a bit to generous, if for whatever reason someone calls their component `e2e.ts` , it would not be compiled right? Is there a chance we can verify if the file content has Stencil primitivis for testing?

Reply:
hmm we could, although this change is pretty much just restoring what the behavior was before #4315. Maybe it's a question of documenting that files which are named like this for some reason wouldn't be emitted?

Reply:
Correct me if I am wrong but before #4315 we didn't had this whole `if` block. What is the intention of including `emitFilePath.includes('/e2e.')` and `emitFilePath.includes('/spec.')`? They don't match with common naming patterns afaics. I'ld suggest maybe only:

```ts
if (
  emitFilePath.includes('.e2e.ts') ||
  emitFilePath.includes('.spec.ts') ||
  emitFilePath === 'e2e.d.ts' ||
  emitFilePath === 'spec.d.ts'
) {
  // we don't want to write these to disk!
  return;
}
```

?

Reply:
Yeah so we didn't have this block, but we did something similar elsewhere - here's the relevant diff: https://github.com/ionic-team/stencil/pull/4315/files#diff-641e8e3ba21b8a7166c9350fe42c34c2b33ccbc220173ecdda7b451da8a423ba

Reply:
I just pulled in the `/spec.` thing because that was there in the old impl - I guess maybe if you had `src/components/my-component/my-component.tsx` maybe you could have `spec.ts` in the same directory?

Reply:
Oh now I see it. Then it would make sense to have this, no?
```ts
if (
  emitFilePath.endsWith('e2e.ts') ||
  emitFilePath.endsWith('spec.ts') ||
  emitFilePath.endsWith('e2e.d.ts') ||
  emitFilePath.endsWith('spec.d.ts')
) {
  // we don't want to write these to disk!
  return;
}
```
Above would catch both cases even though it would surprise me if folks actually used `spec.ts` as test file names and would expect Stencil to detect this as a test file. Anyway, very minor nit, feel free to move forward with what makes sense.

Reply:
yeah, I think that makes sense to me, I'll adjust it a bit!

I may also try to get a little regression test for this using one of our end-to-end projects